pontoon/machinery/utils.py

```diff
@@ -76,10 +76,21 @@ def get_concordance_search_data(text, locale):
     search_results = (
         base.models.TranslationMemoryEntry.objects.filter(search_query)
         .values("source", "target")
-        .annotate(project_names=ArrayAgg("project__name", distinct=True))
+        .annotate(
+            project_names=ArrayAgg(
+                "project__name", ordering=["project__disabled", "-project__priority"]
+            )
+        )
         .distinct()
     )
 
+    # We need to remove duplicates manually - ArrayAgg does not support using distinct=True
+    # in combination with ordering.
+    for s in search_results:
```

Comment:
Do we have to remove duplicates before the pagination? I think we can move this after the pagination is applied to reduce the number of processed rows.

Reply:
What jotes says, and also, would it be useful to just shell out the project names into a separate query and order there? So, first get project IDs with distinct, and then query with ordering.

Maybe also too complicated, I'd make the call depending on how many duplicates there are in practice.

Reply:
@Pike Not sure I understand your part. I'd annotate distinct project IDs (instead of ordered project names) in the `search_results` QuerySet? And then I'd order them and convert them to project names in the loop, using the ordered list of project name values?

Reply:
Right. I noticed that I couldn't just write out what I had in mind, but the gist of it was "do two queries, one for distinct, and one for ordering, and then stitch the results together in a loop"

Reply:
Updated.

I've also included a CSS fix to reveal all project names on hover even if the user is logged out or the project is read-only.